draft-ietf-mls-protocol.md

```diff
@@ -944,23 +947,32 @@ Given these inputs, the derivation of secrets for an epoch
 proceeds as shown in the following diagram:
 
 ~~~~~
-               init_secret_[n-1] (or 0)
-                     |
-                     V
-update_secret -> HKDF-Extract = epoch_secret
-                     |
-                     +--> Derive-Secret(., "app", GroupState_[n])
-                     |    = application_secret
-                     |
-                     +--> Derive-Secret(., "confirm", GroupState_[n])
-                     |    = confirmation_key
-                     |
-                     V
-               Derive-Secret(., "init", GroupState_[n])
-                     |
-                     V
-               init_secret_[n]
-~~~~~
+                  init_secret_[N-1] (or 0)
+                        |
+                        V
+update_secret_[N-1] --> HKDF-Extract = epoch_secret_[N]
+                        |
+                        +--> Derive-Secret(., "app",
+                        |                  GroupState_[N].transcript_hash)
```

Comment:
I would prefer that we not make this change, i.e., that we set the context to the serialized group state.  Changing from the group state to the transcript hash has bigger implications than just changing how the hashes are arranged.  That is, let's do #107 in this PR, and #90 later.  Note that this would also obviate the OPEN ISSUE below.